orchagent/portsorch.cpp

```diff
@@ -1728,12 +1730,34 @@ void PortsOrch::doPortTask(Consumer &consumer)
                                     continue;
                                 }
                             }
-                        }
-                        else
-                        {
-                            SWSS_LOG_ERROR("Failed to get current speed for port %s", alias.c_str());
-                            it++;
-                            continue;
+
+                            if (setPortSpeed(p.m_port_id, speed))
+                            {
+                                SWSS_LOG_NOTICE("Set port %s speed to %u", alias.c_str(), speed);
+                                p.m_speed = speed;
+                            }
+                            else
+                            {
+                                SWSS_LOG_ERROR("Failed to set port %s speed to %u", alias.c_str(), speed);
```

Comment:
Just thinking if we need to set the admin status to `old_admin_state_up` here if setPortSpeed failed.

Reply:
Previous implementation relied on the fact that on each change the entire PORT table entry(speed, admin, fec etc.) is pushed to orchagent even if only one field changed(e.g. speed).
So, if port speed failed, on next iteration we will set admin status to correct DB status from DB.